### DISCUSSION TITLE
model used for swe-bench

### DISCUSSION THREAD

**Discussion by rezzie-rich** [ComposioHQ/composio#841] [Announcements and Release]
Which LLM was used to achieve 48% on swe-bench? was it gpt4o or sonnet?

**Comment by eeshsaxena** [comment]
for swe-bench type tasks the results vary a lot based on which subset (full vs verified) and the scaffolding around the model, not just the raw model itself.

rough ranking for coding/agent tasks as of mid-2026:
- claude sonnet 3.5/3.7 - usually leads on verified swe-bench
- gpt-4o - solid baseline, bit behind claude on code specifically
- deepseek-v3 - surprisingly strong, good if you want open weights

for local models that handle tool calling well (you need that for agentic tasks):
- qwen2.5-coder-32b is probably the best open option right now
- deepseek-coder-v2 is close
- llama-3.1-70b if you need something that runs on consumer hardware

the 48% number is usually with claude or gpt-4o plus a well-tuned agent loop. the scaffolding and context management matter as much as the model itself.
